### PR TITLE
Handle "profile updated" push payloads.

### PR DESCRIPTION
**Pull request by eoger**
Connects to https://github.com/mozilla/application-services/issues/1418.
Let's see if we're happy with the API before committing to handling more push events.

### CODE REVIEW

File: components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/AccountEvent.kt [hunk 1]
```diff
@@ -16,6 +16,8 @@ val <T> T.exhaustive: T
 sealed class AccountEvent {
     // A tab with all its history entries (back button).
     class TabReceived(val from: Device?, val entries: Array<TabHistoryEntry>) : AccountEvent()
+    // A notification that the profile has been updated.
+    class ProfileUpdated() : AccountEvent()
```
Comment (eoger) [inline components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/AccountEvent.kt]: @grigoryk would you like `ProfileUpdated` to contain the updated profile data or should it just stay as a hint that you should fetch the profile?

File: components/fxa-client/examples/devices_api.rs [hunk 2]
```diff
@@ -104,6 +104,7 @@ fn main() -> Result<(), failure::Error> {
                             };
                             webbrowser::open(&tab.url).unwrap();
                         }
+                        _ => {} // Ignore
```
Comment (eoger) [inline components/fxa-client/examples/devices_api.rs]: @grigoryk This unrelated change means that `FirefoxAccount.pollDeviceCommands` (which returns `Array<AccountEvent>`) could theoretically (but will never) return other types of account events such as `ProfileUpdated`. Should we rename this method to `pollAccountEvents` to make it sound more correct?

File: components/fxa-client/src/lib.rs [hunk 3]
```diff
@@ -206,8 +187,8 @@ impl FirefoxAccount {
 }
 
 pub enum AccountEvent {
-    // In the future: ProfileUpdated etc.
     TabReceived((Option<Device>, SendTabPayload)),
+    ProfileUpdated, // TODO: Should we add the profile data here or let the app do it itself?.
```
Comment (eoger) [inline components/fxa-client/src/lib.rs]: See https://github.com/mozilla/application-services/pull/1420/files#r303988532.

**Comment by rfk** [comment]
> This unrelated change means that FirefoxAccount.pollDeviceCommands (which returns 
> Array<AccountEvent>) could theoretically (but will never) return other types of account
> events such as ProfileUpdated

Alternately, I wonder if we can "unflatten" them so that we have `ProfileUpdated` and `DeviceCommandReceived` events, and some higher-level component is responsible for looking inside each `DeviceCommandReceived` event and transforming it into a `TabReceivedCommand` or potentially other kinds of command data in the future.

File: components/fxa-client/src/push.rs [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        let payload = serde_json::from_str(payload)?;
+        match payload {
+            PushPayload::CommandReceived(CommandReceivedPushPayload { index, .. }) => {
+                if cfg!(target_os = "ios") {
+                    self.fetch_device_command(index).map(|cmd| vec![cmd])
+                } else {
+                    self.poll_device_commands()
+                }
+            }
+            PushPayload::ProfileUpdated => Ok(vec![AccountEvent::ProfileUpdated]),
```
Comment (rfk) [inline components/fxa-client/src/push.rs]: This could be a good opportunity to clear our locally-cached profile info, or mark it as stale, or some other action which will cause us to automatically refresh it from cache the next time it is requested.

**Comment by eoger** [comment]
We can revisit this later when Fenix wants this.
